### DISCUSSION TITLE
graph in newest release

### DISCUSSION THREAD

**Discussion by shortwavesurfer2009** [christianrowlands/android-network-survey#29] [General]
What determines where the graph starts for cell signals? Mine starts at -32dbm which seems excessively high for real world use. Personally i have never seen anything stronger than a -75 and wonder if the graph should be moved closer to that level.

Edit: my thought is to do 13. Example -75, -88, -101, -114, and -127 for the lines on the y axis. If you can add a 6th line you could do -75, -85, -95, -105, -115, and -125

**Comment by christianrowlands** [comment]
For the graph ranges I went with the 3GPP specification defined range. For NR, the range is -31 dBm to -156 dBm (I bumped it down one to -32 to make the graph intervals avoid decimals).

I debated going smaller, but if someone was walking down a tower using the signal graph, I figured it would be nice if the graph went up to the maximum value that the phone is able to report. You are correct that in everyday life you probably won't see values like this. I created the Wi-Fi and Bluetooth charts first, and I actually created them because I had a Wi-Fi network show up that I was not expecting to be there and wanted to figure out where it was coming from. Therefore, I had walking down the signal source in mind when I started working on the cellular chart.

I think if you did want to walk down a cellular tower you could rely on just looking at the number in the UI. I don't think the chart would help all that much because it changes so much more slowly than walking down a Wi-Fi AP. I will update the ranges to be more real world (similar to how I have the signal indicator that surrounds the number where it maxes out at a much more reasonable value -70 for NR).

**Reply by christianrowlands** [reply]
@shortwavesurfer2009  I update the ranges in the latest release 1.19. Take a look and let me know if those ranges are better.

**Comment by shortwavesurfer2009** [comment]
Oh yeah, that is much better.

Sent from Proton Mail mobile

-------- Original Message --------
